design-proposals/volume-update-strategy.md

```diff
@@ -0,0 +1,218 @@
+# Overview
+This design proposal extends KubeVirt APIs to enable the storage migration between PVCs.
+
+## Motivation
+In some circumstances, the guest owner may want to have the option of switching the storage provider or class for a group of volumes used by an active VM. Currently, the VM must be turned off and recreated, causing service disruptions and downtime.
+
+
+## Goals
+Design and develop an API to live update and migrate PVCs.
+  * The volume migration needs to be declarative and [GitOps](https://kubevirt.io/user-guide/operations/gitops/) compatible
+  * Volume migration implies an update of the volumes used by the VM. Hence, we want to trigger the storage migration on a VM update.
+  * The VM spec updates needs to be triggered by the user and not to be implicitly updated by the KubeVirt controller
+
+ ## Non  Goals
+*  This feature is not directly aimed at live migrating the VM using RWO volumes. Hence, the API covers and refers only to the volume update.
+*  This design doesn't cover migration between storage classes, but rather implements basic APIs that can be used by overlying tooling.
```

Comment:
I would maybe leave this out?  Migrating between storageclasses is one of the main goals even if it is not a detail that is important to the implementation

Reply:
Here, I wanted to mentioned that this API doesn't deal with storage classes. We started this work by tring to design an API directly using storage classes and then remove it. I'd like to mention to the user that they don't need to expect any storage class notion with this feature. Does this make sense?